platform/windows/detect.py

```diff
@@ -763,6 +759,11 @@ def configure_mingw(env: "SConsEnvironment"):
         env.Append(CCFLAGS=san_flags)
         env.Append(LINKFLAGS=san_flags)
 
+    if env["use_llvm"] and os.name == "nt" and methods._colorize:
```

Comment:
Do we need `os.name == "nt"` in the check? MSYS2 and cross building should support escape sequences as well.

Reply:
The `-fansi-escape-codes` flag only takes effect on Windows (LLVM/Clang always use ANSI color on Linux), and last time I checked color diagnostics works out of the box when building on Linux without these changes.

Re MSYS2, it should be no different when using a native build of python (e.g. from the CLANG64 environment).
If you use the msys2 build of python from the MSYS environment, I suspect things may be a bit broken, for example `use_windows_spawn_fix` also checks `os.name != "nt"`.